device-registry/src/main/scala/org/genivi/sota/device_registry/Boot.scala

```diff
@@ -78,6 +83,20 @@ object Boot extends BootApp with Directives with BootMigrations
       new HealthResource(db, versionMap).route
     }
 
+  private def dbOp[T<: BusMessage](msg: T) = {
+    import org.genivi.sota.data.DeviceSearchCommon._
+    val updateSpecMsg = msg.asInstanceOf[UpdateSpec]
+    db.run(for {
+      device <- DeviceRepository.findByUuid(updateSpecMsg.device)
```

Comment:
Why are you looking up the `device`? To find out if it is missing? `setDeviceStatus` will do that for you. If it is because you want to find the UUID then `updateSpecMsg.device` is the UUID already.